src/_gettsim/transfers/arbeitsl_geld_2/kindergelduebertrag.py

```diff
@@ -0,0 +1,113 @@
+"""Module for the calculation of the Kindergeldübertrag."""
+
+import numpy as np
+
+from _gettsim.shared import join_numpy, policy_info
+
+aggregate_by_p_id_kindergeldübertrag = {
+    "kindergeld_übertrag_m": {
+        "p_id_to_aggregate_by": "p_id_kindergeld_empf",
```

Comment:
Just thinking of a weird case here: what if Kindergeldempfänger is outside the BG? E.g. ex-partner. Is that even allowed? Do we have/need a check?

Reply:
The concept of Kindergeldübertrag is that the Kindergeldempfänger is in a different BG than the child (and hence there is Kindergeldübertrag). Are you thinking about different `fg`s?

I.e.
| p_id | hh_id | fg_id | bg_id | p_id_kindergeld_empf | p_id_elternteil_1 | p_id_elternteil_2 | kind  |
|------|-------|-------|-------|----------------------|-------------------|-------------------|-------|
| 0    | 0     | 0     | 0     | -1                   | -1                | -1                | false |
| 1    | 0     | 1     | 1     | -1                   | -1                | -1                | false |
| 2    | 0     | 0     | 2     | 1                    | 0                 | 1                 | true  |

I'd argue that this is not possible because then, the child should count towards the `fg` of `p_id==1`. But we don't have a test for that, we don't account for separated couples in the same household when creating `fg_id`.

Reply:
Yes, I meant fg's.

Precondition for fg is living in the same household. I just wonder whether it is possible that the other parent gets the Kindergeld. And if not (what I'd kinda expect), whether we check that as a condition on the data.

Reply:
I'll create an issue on the creation of `fg_id`. Intuitively, the child should form an `fg` with the parent that gets the Kindergeld (but could be something else of course). It's not clear to me what GETTSIM is currently doing in that regard.

Reply:
I am pretty sure the fg conditions on being part of the same household. My main question is whether Kindergeld does the same when parents are separated.

Reply:
Kindergeld does not affect `fg_id` currently

Reply:
Let me rephrase, that was not clear: 

My main question is whether `kindergeldempfänger` conditions that parent being part of the same household when parents are separated — and we are talking about means-tested transfers (ofc parents receive kindergeld for kids at college even though they may not be in the same household).

*(I am just worried that the machinery here depends implicitly on this assumption being true, would like to make it explicit)*

Reply:
Okay so my point was that the creation of `fg_id` in case parents have a child, are separated **but live in the same household** should react to who claims Kindergeld.

Your point is that if parents **live in different households** `p_id_kindergeldempfänger` should be compatible with the place of living of the child (if such a requirement exists, presumably it does) if one of the involved individuals gets transfers?

> (I am just worried that the machinery here depends implicitly on this assumption being true, would like to make it explicit)

We create neither of the variables endogenously. I'm not sure to which extent creating a consistency check here would create a circle in the DAG.

Reply:
> Your point is that if parents **live in different households** `p_id_kindergeldempfänger` should be compatible with the place of living of the child (if such a requirement exists, presumably it does) if one of the involved individuals gets transfers?

Exactly.

> > (I am just worried that the machinery here depends implicitly on this assumption being true, would like to make it explicit)
> 
> We create neither of the variables endogenously. I'm not sure to which extent creating a consistency check here would create a circle in the DAG.

Would be a check pre-DAG,  just whether the data passed conforms to expectations.

Reply:
But how do we check pre-DAG whether the involved individuals get transfers?

If I understood it correctly, if all involved individuals don't have their SGB II Bedarf gedeckt, we want to allow for

| p_id | hh_id | fg_id | bg_id | p_id_kindergeld_empfänger | kind  |
|------|-------|-------|-------|---------------------------|-------|
| 0    | 0     | 0     | 0     | -1                        | false |
| 1    | 0     | 0     | 0     | 0                         | true  |
| 2    | 1     | 1     | 1     | -1                        | false |

but we don't want to allow for / it would be inconsistent if

| p_id | hh_id | fg_id | bg_id | p_id_kindergeld_empfänger | kind  |
|------|-------|-------|-------|---------------------------|-------|
| 0    | 0     | 0     | 0     | -1                        | false |
| 1    | 0     | 0     | 0     | 2                         | true  |
| 2    | 1     | 1     | 1     | -1                        | false |

But if all adults have their SGB II Bedarf gedeckt, the latter situation would be perfectly fine.

Reply:
Fine, let's not get carried away here. It may be more about `alleinerziehend` status (or, more precisely, having `unterhaltsanspruch`) and `kindergeld`, basically it is baked in there already (deduction of 1/2 `kindergeld` from the aluminum payment). We should still think of a check for that, but no need to do it here. 

As always, closing an issue by thinking through it opens at least three new ones... 😂